.github/workflows/ci.yaml

```diff
@@ -96,88 +96,94 @@ jobs:
         run: go test -race ./... -timeout 2m
 
   cli:
-    runs-on: ubuntu-latest
-    timeout-minutes: 3
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
 
-    services:
-      postgres:
-        image: postgres
-        env:
-          POSTGRES_PASSWORD: postgres
-        options: >-
-          --health-cmd pg_isready
-          --health-interval 2s
-          --health-timeout 5s
-          --health-retries 5
-        ports:
-          - 5432:5432
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 10
 
     steps:
+      - uses: ikalnytskyi/action-setup-postgres@v6
+        with:
+          database: river_dev
+          password: postgres
+
       - uses: actions/setup-go@v5
         with:
           go-version: "stable"
           check-latest: true
 
+      - name: Setup GOBIN (Linux)
+        if: runner.os == 'Linux'
+        run: |
+          echo "GOBIN=$(go env GOPATH)/bin" >> $GITHUB_ENV
+          echo "PATH=$(go env GOPATH)/bin:$PATH" >> $GITHUB_ENV
+        shell: bash
+
+      - name: Setup GOBIN (Windows)
+        if: runner.os == 'Windows'
+        run: |
+          $gobin = "$(go env GOPATH)\bin"
+          echo "GOBIN=$gobin" >> $env:GITHUB_ENV
+          echo $gobin | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+        shell: pwsh
```

Comment:
It got a little messy trying to do this in a reliably cross-platform way using `bash`, so I decided it was easier to split out the windows-specific form.